pkg/controller/networkpolicy/clusternetworkpolicy.go

```diff
@@ -254,13 +258,158 @@ func (n *NetworkPolicyController) deleteNamespace(old interface{}) {
+	serviceAccount := obj.(*v1.ServiceAccount)
+	klog.V(2).InfoS("Processing ServiceAccount ADD event", "ServiceAccountNS", serviceAccount.Namespace, "ServiceAccountName", serviceAccount.Name, "ServiceAccountLabels", serviceAccount.Labels)
+	affectedACNPs := n.filterACNPsByServiceAccount(serviceAccount)
+	for _, acnp := range affectedACNPs {
+		n.reprocessCNP(acnp, false)
+	}
+}
+
+// updateServiceAccount receives ServiceAccount UPDATE events and triggers all
+// ClusterNetworkPolicies that refer to this ServiceAccount in `serviceAccounts`
+// field to be re-processed.
+func (n *NetworkPolicyController) updateServiceAccount(oldObj, curObj interface{}) {
+	defer n.heartbeat("updateServiceAccount")
+	oldServiceAccount, curServiceAccount := oldObj.(*v1.ServiceAccount), curObj.(*v1.ServiceAccount)
+	if reflect.DeepEqual(oldServiceAccount.Labels, curServiceAccount.Labels) {
+		klog.V(2).InfoS("Labels of ServiceAccount didn't change, so skip ACNP reprocess", "ServiceAccountNS", curServiceAccount.Namespace, "ServiceAccountName", curServiceAccount.Name, "ServiceAccountLabels", curServiceAccount.Labels)
+		return
+	}
+	klog.V(2).InfoS("Processing ServiceAccount UPDATE event", "ServiceAccountNS", curServiceAccount.Namespace, "ServiceAccountName", curServiceAccount.Name, "OldServiceAccountLabels", oldServiceAccount.Labels, "CurServiceAccountLabels", curServiceAccount.Labels)
+	affectedACNPsByOldServiceAccount := n.filterACNPsByServiceAccount(oldServiceAccount)
+	affectedACNPsByCurServiceAccount := n.filterACNPsByServiceAccount(curServiceAccount)
+	getACNPNamesSet := func(acnps []*crdv1alpha1.ClusterNetworkPolicy) sets.String {
+		nameSet := sets.String{}
+		for _, acnp := range acnps {
+			nameSet.Insert(acnp.Name)
+		}
+		return nameSet
+	}
+	oldACNPNameSet := getACNPNamesSet(affectedACNPsByOldServiceAccount)
+	curACNPNameSet := getACNPNamesSet(affectedACNPsByCurServiceAccount)
+	affectedACNPNameSet := utilsets.SymmetricDifferenceString(oldACNPNameSet, curACNPNameSet)
+	for _, acnp := range affectedACNPsByOldServiceAccount {
+		if affectedACNPNameSet.Has(acnp.Name) {
+			n.reprocessCNP(acnp, false)
+		}
+	}
+	for _, acnp := range affectedACNPsByCurServiceAccount {
+		if affectedACNPNameSet.Has(acnp.Name) {
+			n.reprocessCNP(acnp, false)
+		}
+	}
+}
+
+// deleteNamespace receives ServiceAccount DELETE events and triggers all
+// ClusterNetworkPolicies that refer to this ServiceAccount in `serviceAccounts`
+// field to be re-processed.
+func (n *NetworkPolicyController) deleteServiceAccount(old interface{}) {
+	serviceAccount := old.(*v1.ServiceAccount)
+	defer n.heartbeat("deleteServiceAccount")
+	klog.V(2).InfoS("Processing ServiceAccount DELETE event", "ServiceAccountNS", serviceAccount.Namespace, "ServiceAccountName", serviceAccount.Name, "ServiceAccountLabels", serviceAccount.Labels)
+	affectedACNPs := n.filterACNPsByServiceAccount(serviceAccount)
+	for _, acnp := range affectedACNPs {
+		n.reprocessCNP(acnp, false)
+	}
```

Comment:
Just for troubleshooting purposes, could it be useful to add a log line stating that delete service account processing is complete (same for add). In this way we will be able to figure out from logs if the routine did not select any network policy or whether instead the network policies were not processed.

This is a  minor comment. No need to address it.

Reply:
Thanks. Yeah adding a log at the bottom line could help us know the process is complete. And regarding this:

> if the routine did not select any network policy or whether instead the network policies were not processed

I guess the log inside reprocessCNP is enough?